x-pack/plugins/upgrade_assistant/server/lib/reindexing/reindex_service.ts

```diff
@@ -347,6 +347,11 @@ export const reindexServiceFactory = (
       await esClient.indices.open({ index: indexName });
     }
 
+    const flatSettings = await actions.getFlatSettings(indexName);
```

Comment:
It's not clear to me why this was removed on `master`, but it seems like it is a valid check.

Reply:
Hmm, good point! It may have been removed in error.